csrc/mmdeploy/codebase/mmpose/cuda/topdown_affine.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include "mmdeploy/codebase/mmpose/topdown_affine.h"
+
+#include <vector>
+
+#include "ppl/cv/cuda/crop.h"
+#include "ppl/cv/cuda/resize.h"
+
+namespace mmdeploy::mmpose {
+
+class CropAndResizeCudaImpl : public CropAndResize {
+ public:
+  CropAndResizeCudaImpl() = default;
+
+  Result<void> apply(const std::vector<float>& bbox, std::vector<float>& c, std::vector<float>& s,
+                     const std::vector<int>& output_size, const Tensor& src, Tensor& dst) override {
+    OUTCOME_TRY(auto img, MakeAvailableOnDevice(src, device_, stream()));
+    stream().Wait().value();
+
+    // x, y, w, h
+    std::vector<int> roi = {int(bbox[0]), int(bbox[1]), (int)bbox[2], (int)bbox[3]};
+    // update center
+    c[0] = roi[0] + roi[2] / 2.f;
+    c[1] = roi[1] + roi[3] / 2.f;
+    roi[2] = std::min(roi[2], (int)src.shape(2) - roi[0]);
+    roi[3] = std::min(roi[3], (int)src.shape(1) - roi[1]);
+    auto scale = std::min(output_size[0] / 200.f / s[0], output_size[1] / 200.f / s[1]);
+    int rw = roi[2] * scale + 0.5;
+    int rh = roi[3] * scale + 0.5;
+    // update scale
+    s[0] = output_size[0] / 200.f / rw * roi[2];
+    s[1] = output_size[1] / 200.f / rh * roi[3];
+    TensorDesc resized_desc = {
+        .device = device_,
+        .data_type = DataType::kINT8,
+        .shape = {1, rh, rw, 3},
+    };
+    Tensor resized = Tensor(resized_desc);
+
+    TensorDesc desc = {.device = device_,
+                       .data_type = DataType::kINT8,
+                       .shape = {1, output_size[1], output_size[0], 3}};
+    dst = Tensor(desc);
+
+    auto status = ppl::cv::cuda::Resize<uint8_t, 3>(
+        (cudaStream_t)stream().GetNative(), roi[3], roi[2], img.shape(2) * img.shape(3),
+        img.data<uint8_t>() + roi[1] * img.shape(2) * img.shape(3) + roi[0] * img.shape(3), rh, rw,
+        rw * 3, resized.data<uint8_t>(), ppl::cv::InterpolationType::INTERPOLATION_LINEAR);
+
+    cudaMemsetAsync(dst.data<uint8_t>(), 0, dst.byte_size(), (cudaStream_t)stream().GetNative());
+    std::vector<int> droi = {int(dst.shape(2) - resized.shape(2)) / 2,
+                             int(dst.shape(1) - resized.shape(1)) / 2, (int)resized.shape(2),
+                             (int)resized.shape(1)};
+
+    cudaMemcpy2DAsync(dst.data<uint8_t>() + droi[1] * dst.shape(2) * dst.shape(3) + droi[0] * 3,
+                      dst.shape(2) * dst.shape(3), resized.data<uint8_t>(), rw * 3, rw * 3, rh,
+                      cudaMemcpyDefault, (cudaStream_t)stream().GetNative());
```

Comment:
1. This is a `Pad` operation. The operation should be named `CropResizePad`. 
2. Compute the parameter for Crop (a rect), Resize (a size) and Pad (tlbr) somewhere else.
3. `ppl::cv::Resize` supports strided output (which corresponds to a RoI), clear the dst buffer and resize directly onto the RoI. The `cudaMemcpy2DAsync` can be saved.